### PR TITLE
Context

### PR DESCRIPTION
**Pull request by Thomzoy**
Add the option to store document-level extensions when using processing pipelines

- Add a `ConceptAdder` pipe with a `set_extension()` 
- Modify each pipe to add a `context` argument
- Checking SpaCy version when using a context (needs `spacy>=3.2`)

- [x] If this PR is a bug fix, the bug is documented in the test suite.
- [x] Changes were documented in the changelog (pending section).
- [x] If necessary, changes were made to the documentation (eg new pipeline).

### CODE REVIEW

File: edsnlp/processing/distributed.py [hunk 1]
```diff
@@ -60,6 +67,11 @@ def pipe(
         A Pyspark or Koalas DataFrame with a `note_id` and `note_text` column
     nlp : Language
         A spaCy pipe
+    context : List[str]
+        A list of column to add to the generated SpaCy document as an extension.
```
Comment (percevalw) [inline edsnlp/processing/distributed.py:71]: Typo

```suggestion
        A list of columns to add to the generated SpaCy document as an extension.
```

File: edsnlp/processing/parallel.py [hunk 2]
```diff
@@ -61,6 +63,11 @@ def pipe(
         A pandas DataFrame with a `note_id` and `note_text` column
     nlp : Language
         A spaCy pipe
+    context : List[str]
+        A list of column to add to the generated SpaCy document as an extension.
```
Comment (percevalw) [inline edsnlp/processing/parallel.py:67]: Typo
```suggestion
        A list of columns to add to the generated SpaCy document as an extension.
```

File: edsnlp/processing/simple.py [hunk 3]
```diff
@@ -56,26 +71,32 @@ def _flatten(list_of_lists: List[List[Any]]):
 def _pipe_generator(
     note: pd.DataFrame,
     nlp: Language,
+    context: List[str] = [],
     additional_spans: Union[List[str], str] = "discarded",
     extensions: ExtensionSchema = [],
-    batch_size: int = 1000,
+    batch_size: int = 50,
```
Comment (percevalw) [inline edsnlp/processing/simple.py:77]: Should this be part of the PR ?

File: edsnlp/processing/simple.py [hunk 4]
```diff
@@ -168,6 +190,11 @@ def pipe(
         A pandas DataFrame with a `note_id` and `note_text` column
     nlp : Language
         A spaCy pipe
+    context : List[str]
+        A list of column to add to the generated SpaCy document as an extension.
```
Comment (percevalw) [inline edsnlp/processing/simple.py:194]:
```suggestion
        A list of columns to add to the generated SpaCy document as an extension.
```

File: edsnlp/processing/wrapper.py [hunk 5]
```diff
@@ -26,6 +27,11 @@ def pipe(
         A pandas/pyspark/koalas DataFrame with a `note_id` and `note_text` column
     nlp : Language
         A spaCy pipe
+    context : List[str]
+        A list of column to add to the generated SpaCy document as an extension.
```
Comment (percevalw) [inline edsnlp/processing/wrapper.py:31]:
```suggestion
        A list of columns to add to the generated SpaCy document as an extension.
```

**Review by percevalw — APPROVED** [review]
Great, thanks !

File: edsnlp/processing/simple.py (hunk 3 above)
Comment (Thomzoy) [reply edsnlp/processing/simple.py:77]: It isn't related but it still seems that 1000 is way too big, especially for medical texts which are usually pretty long

**Pull request merged**